core/src/main/java/io/cucumber/core/options/CommandlineOptionsParser.java

```diff
@@ -89,7 +91,11 @@ private RuntimeOptionsBuilder parse(List<String> args) {
                 URI parse = GluePath.parse(gluePath);
                 parsedOptions.addGlue(parse);
             } else if (arg.equals("--tags") || arg.equals("-t")) {
-                parsedOptions.addTagFilter(removeArgFor(arg, args));
+                try {
+                    parsedOptions.addTagFilter(TagExpressionParser.parse(removeArgFor(arg, args)));
+                } catch (TagExpressionException tee) {
+                    throw new IllegalArgumentException(tee);
```

Comment:
We'll have to look at this. Currently the problem is that it doesn't report back where the problem came from. But this is true for all other CLI arguments as well.

Reply:
It may be easier if we have a shared session to discuss the options.